### PR TITLE
PCRM-3845-Test-Coverage

### PR DESCRIPTION
**Pull request by premnhs**

### CODE REVIEW

File: test/integrationtests/step_impl/lr_15_lambda_cdd_file.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+LR_10_STATE_FUNCTION_ARN = AWS_RESOURCE["lr_10_sfn_arn"]["value"]
+LR_22_BUCKET = AWS_RESOURCE["lr_22_bucket"]["value"]
+LR_13_BUCKET = AWS_RESOURCE["lr_13_bucket"]["value"]
+DYNAMODB_DEMOG = AWS_RESOURCE["demographic_table"]["value"]
+DYNAMODB_INFLIGHTS = AWS_RESOURCE["inflight_table"]["value"]
+DYNAMODB_JOBS = AWS_RESOURCE["jobs_table"]["value"]
+DYNAMODB_JOBSTATS = AWS_RESOURCE["jobs_Status_table"]["value"]
+
+
+
```
Comment (josh-scargill-nhsd) [inline test/integrationtests/step_impl/lr_15_lambda_cdd_file.py:36]: 2 new lines above functions not 3

File: test/integrationtests/step_impl/lr_15_lambda_cdd_file.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    upload_pds_data_set02 = os.path.join(DATA, "LR_15/" + pds_data_set02)
+    upload_pds_data_set03 = os.path.join(DATA, "LR_15/" + pds_data_set03)
+
+    s3 = dev.client("s3", REGION_NAME)
+
+    try:
+        s3.upload_file(upload_pds_data, MOCK_PDS_DATA, pds_api_data_file)
+        s3.upload_file(upload_pds_data_set1, LR_22_BUCKET, pds_data_set01)
+        s3.upload_file(upload_pds_data_set02, LR_22_BUCKET, pds_data_set02)
+        s3.upload_file(upload_pds_data_set03, LR_22_BUCKET, pds_data_set03)
```
Comment (josh-scargill-nhsd) [inline test/integrationtests/step_impl/lr_15_lambda_cdd_file.py:55]: this is 12 lines that could easily be 4:
```
s3 = dev.client("s3", REGION_NAME)

try:
    s3.upload_file(upload_pds_data, MOCK_PDS_DATA, os.path.join(DATA, lr_15_path + "pds_api_data.csv"))
    s3.upload_file(upload_pds_data_set1, LR_22_BUCKET, os.path.join(DATA, "LR_15/A82023.csv"))
    s3.upload_file(upload_pds_data_set02, LR_22_BUCKET, os.path.join(DATA, "LR_15/Y123451.csv"))
    s3.upload_file(upload_pds_data_set03, LR_22_BUCKET, os.path.join(DATA, "LR_15/Y123452.csv"))
```

File: test/integrationtests/step_impl/lr_02_lambda_val_parse.py [hunk 3]
```diff
@@ -197,3 +202,15 @@ def readfile_in_s3_failed_invalid_item(search_word):
...
+def assert_fileloaded_in_s3_pass_folder():
+    s3 = dev.client("s3", REGION_NAME)
+    expected_filename = "pass/" + destination_filename
+    result = s3.list_objects(
+        Bucket=LR_01_BUCKET, Prefix=expected_filename
+    )
+    if result.get("Contents") != None:
+        for o in result.get("Contents"):
+            actual_filename = o.get("Key")
+            assert expected_filename == actual_filename, "destination file not found"
```
Comment (jamesnunn) [inline test/integrationtests/step_impl/lr_02_lambda_val_parse.py:216]:
```for o in result.get("Contents", []):
    actual_filename = o.get("Key")
    assert expected_filename == actual_filename, "destination file not found"
else:
    raise AssertionErrororSomeOtherError('No items in contents')
```

File: test/integrationtests/step_impl/lr_15_lambda_cdd_file.py (hunk 2 above)
Comment (premnhs) [reply test/integrationtests/step_impl/lr_15_lambda_cdd_file.py:55]: thanks Josh its done..but it has to be done the below way bcoz  it has to be : 'upload_file(Filename, Bucket, Key)'
`    try:
        s3.upload_file(os.path.join(DATA, lr_15_path + "pds_api_data.csv"), MOCK_PDS_DATA, "pds_api_data.csv")
        s3.upload_file(os.path.join(DATA, "LR_15/A82023.csv"), LR_22_BUCKET, "LR_15/A82023.csv" )
        s3.upload_file(os.path.join(DATA, "LR_15/Y123451.csv"), LR_22_BUCKET, "LR_15/Y123451.csv")
        s3.upload_file(os.path.join(DATA, "LR_15/Y123452.csv"), LR_22_BUCKET, "LR_15/Y123452.csv")`

File: test/integrationtests/specs/lr_15_lambda_cdd_file.spec [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Tests for LR-15 to generate consolidated csv file with the respective gppractive comparision records 
+-------------------------------------------------------------------------------------------------------
+* setup steps: clear all files in LR_01 bucket folders and dynamodb Inflight table
+
+
+## test to ensure consolidated demographic differences file is generated succesfully
+------------------------------------------------------------------------------------
+* setup steps: clear all files in mock pds data and lr_22 buckets
```
Comment (HarveyNHS) [inline test/integrationtests/specs/lr_15_lambda_cdd_file.spec:8]: You could move this up to line 4 and make this as a before step like with line 3

File: test/integrationtests/step_impl/lr_15_lambda_cdd_file.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        s3.upload_file(os.path.join(DATA, "LR_15/A82023.csv"), LR_22_BUCKET, "LR_15/A82023.csv" )
+        s3.upload_file(os.path.join(DATA, "LR_15/Y123451.csv"), LR_22_BUCKET, "LR_15/Y123451.csv")
+        s3.upload_file(os.path.join(DATA, "LR_15/Y123452.csv"), LR_22_BUCKET, "LR_15/Y123452.csv")
+        time.sleep(5)
+        Messages.write_message("All PDS data related files Uploaded Successful")
+
+    except FileNotFoundError:
+        Messages.write_message("File not found")
+        raise
+
```
Comment (HarveyNHS) [inline test/integrationtests/step_impl/lr_15_lambda_cdd_file.py:51]: Did you use 'black' to format the document?

**Review by HarveyNHS — COMMENTED** [review]
Left a couple of comments for you @premnhs

**Review by jamesnunnnhs — APPROVED** [review]

**Review by HarveyNHS — APPROVED** [review]
Approved

**Pull request merged**